presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/json/format/JsonFormatFunctions.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.encoder.json.format;
+
+import io.prestosql.spi.type.TimestampType;
+import io.prestosql.spi.type.TimestampWithTimeZoneType;
+import io.prestosql.spi.type.Type;
+
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.prestosql.spi.type.DateType.DATE;
+import static io.prestosql.spi.type.TimeType.TIME;
+import static io.prestosql.spi.type.TimeWithTimeZoneType.TIME_WITH_TIME_ZONE;
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public final class JsonFormatFunctions
+{
+    public enum FormatType {
+        CUSTOM_DATE_TIME(true),
+        ISO8601(true),
+        RFC2822(true),
+        MILLISECONDS_SINCE_EPOCH(false),
+        SECONDS_SINCE_EPOCH(false);
+
+        private final boolean supportsDate;
+
+        FormatType(boolean supportsDate)
+        {
+            this.supportsDate = supportsDate;
+        }
+
+        public boolean isSupportedType(Type type)
+        {
+            if (type == TIME ||
+                    type == TIME_WITH_TIME_ZONE ||
+                    type instanceof TimestampType ||
+                    type instanceof TimestampWithTimeZoneType) {
+                return true;
+            }
+            return supportsDate && type == DATE;
+        }
+
+        @Override
+        public String toString()
+        {
+            return name().toLowerCase(Locale.ENGLISH).replaceAll("_", "-");
+        }
+    }
+
+    private JsonFormatFunctions() {}
+
+    public static long daysToEpochMillis(long value)
+    {
+        return DAYS.toMillis(value);
```

Comment:
This is subjective, but you could just as well inline.  I think `DAYS.toMillis(value)` reads pretty well.